sharding/data_modeling.rst

```diff
@@ -36,276 +45,40 @@ The following diagram illustrates co-location in the multi-tenant data model. It
-    day date,
-    page_id int,
-    view_count bigint,
-    PRIMARY KEY (day, page_id)
-  );
-
-  SELECT create_distributed_table('daily_page_views', 'page_id');
-
-Precomputing aggregates is called *roll-up*. Notice that distributing both tables by :code:`page_id` co-locates their data per-page. Any aggregate functions grouped per page can run in parallel, and this includes aggregates in roll-ups. We can use PostgreSQL `UPSERT <https://www.postgresql.org/docs/current/static/sql-insert.html#SQL-ON-CONFLICT>`_ to create and update rollups, like this (the SQL below takes a parameter for the lower bound timestamp):
-
-.. code-block:: postgres
-
-  INSERT INTO daily_page_views (day, page_id, view_count)
-  SELECT view_time::date AS day, page_id, count(*) AS view_count
-  FROM page_views
-  WHERE view_time >= $1
-  GROUP BY view_time::date, page_id
-  ON CONFLICT (day, page_id) DO UPDATE SET
-    view_count = daily_page_views.view_count + EXCLUDED.view_count;
-
-Events and Entities
-~~~~~~~~~~~~~~~~~~~
-
-Behavioral analytics seeks to understand users, from the website/product features they use to how they progress through funnels, to the effectiveness of marketing campaigns. Doing analysis tends to involve unforeseen factors which are uncovered by iterative experiments. It is hard to know initially what information about user activity will be relevant to future experiments, so analysts generally try to record everything they can. Using a distributed database like Citus allows them to query the accumulated data flexibly and quickly.
-
-Let's look at a simplified example. Whereas the previous examples dealt with a single events table (possibly augmented with precomputed rollups), this one uses two main tables: users and their events. In particular, Wikipedia editors and their changes:
-
-.. code-block:: postgres
-
-  CREATE TABLE wikipedia_editors (
-    editor TEXT UNIQUE,
-    bot BOOLEAN,
-
-    edit_count INT,
-    added_chars INT,
-    removed_chars INT,
-
-    first_seen TIMESTAMPTZ,
-    last_seen TIMESTAMPTZ
-  );
-
-  CREATE TABLE wikipedia_changes (
-    editor TEXT,
-    time TIMESTAMP WITH TIME ZONE,
-
-    wiki TEXT,
-    title TEXT,
-
-    comment TEXT,
-    minor BOOLEAN,
-    type TEXT,
-
-    old_length INT,
-    new_length INT
-  );
-
-  SELECT create_distributed_table('wikipedia_editors', 'editor');
-  SELECT create_distributed_table('wikipedia_changes', 'editor');
-
-These tables can be populated by the Wikipedia API, and we can distribute them in Citus by the :code:`editor` column. Notice that this is a text column. Citus' hash distribution uses PostgreSQL hashing which supports a number of data types.
+Real-time queries typically ask for numeric aggregates grouped by date or category. Citus sends these queries to each shard for partial results and assembles the final answer on the coordinator node. Queries run fastest when as many nodes contribute as possible, and when no individual node bottlenecks.
```

Comment:
> Queries run fastest when as many nodes contribute as possible, and when no individual node bottlenecks.

This sentence reads a bit awkwardly. What do we intend to say with 'bottlenecks'?